[OpenMP] codegen support for masked combined construct parallel masked taskloop

chandraghale opened the pull request:
Added codegen support for combined masked constructs Parallel masked taskloop. 
Added implementation for EmitOMPParallelMaskedTaskLoopDirective.

Commit:
codegen support for masked combined construct parallel masked taskloop

jdoerfert reviewed (approved):
Looks like the old master directive implementation.

alexey-bataev commented:
Update OpenMPSupport.rst

Commit:
Update OpenMPSupport.rst

chandraghale commented:
> Update OpenMPSupport.rst

Updated OpenMPSupport.rst . Updated the doc for all the related split-ed  PRs ( [PR-121746](https://github.com/llvm/llvm-project/pull/121746) , [121914](https://github.com/llvm/llvm-project/pull/121914) , [121916](https://github.com/llvm/llvm-project/pull/121916)  ) for combined masked construct in this PR only to avoid merge conflict.

alexey-bataev commented:
Also update ReleaseNotes.rst here and in other patches

Commit:
Release Notest update

alexey-bataev reviewed (approved):
LG

chandraghale commented:
> Also update ReleaseNotes.rst here and in other patches

@alexey-bataev release notes updated.  Updated for other patches as well in this PR.

chandraghale committed:
Merge branch 'main' into parallel_masked
